radicle/src/cob/op.rs

```diff
@@ -35,18 +36,31 @@ pub struct Op<A> {
     pub timestamp: clock::Physical,
 }
 
-impl<'a: 'de, 'de, A: serde::Deserialize<'de>> TryFrom<&'a EntryWithClock> for Op<A> {
+pub struct Ops<A>(pub NonEmpty<Op<A>>);
+
+impl<'a, A> TryFrom<&'a EntryWithClock> for Ops<A>
+where
+    for<'de> A: serde::Deserialize<'de>,
+{
     type Error = OpDecodeError;
 
     fn try_from(entry: &'a EntryWithClock) -> Result<Self, Self::Error> {
-        let action = serde_json::from_slice(entry.contents())?;
-
-        Ok(Op {
-            action,
-            author: *entry.actor(),
-            clock: entry.clock().into(),
-            timestamp: entry.timestamp().into(),
-        })
+        let mut clock = entry.clock().into();
+        entry
+            .contents()
+            .clone()
+            .try_map(|op| {
+                let action = serde_json::from_slice(&op)?;
+                let op = Op {
+                    action,
+                    author: *entry.actor(),
+                    clock,
+                    timestamp: entry.timestamp().into(),
+                };
+                clock = clock.tick();
```

Comment:
`clock.tick()` should suffice, as it mutates the internal value as well.

Reply:
Ah, what's the reason for returning `Self` then? I found that a bit confusing

Reply:
I often used it like `New { clock: old.clock.tick() }`, was just convenience

Reply:
I think we can change that to only return but not mutate, since we use the DAG clock now.